sklearn/linear_model/cd_fast.pyx

```diff
@@ -628,7 +631,7 @@ def enet_coordinate_descent_gram(double[:] w, double alpha, double beta,
 @cython.wraparound(False)
 @cython.cdivision(True)
 def enet_coordinate_descent_multi_task(double[::1, :] W, double l1_reg,
-                                       double l2_reg, double[::1, :] X,
+                                       double l2_reg, np.ndarray[double, ndim=2] X,
                                        double[:, :] Y, int max_iter,
```

Comment:
I think `Y` is also impacted. Please write a non regression tests using the `Lasso` class and both readonly input `X` and output `Y` (single output and multiple output regression) to check for all the possible cases.